azure-iot-device/azure/iot/device/common/version_compat.py

```diff
@@ -0,0 +1,37 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module defines functions intended for providing compatibility between
+different versions of Python"""
+
+from six.moves import urllib
+
+
+def urlencode(query, quote_via=urllib.parse.quote_plus, safe=""):
```

Comment:
>urlencode [](start = 4, length = 9)

also a function of this name already exists in urllib right ? so better to rename for us.

Reply:
The idea here is that this is the exact same function as the one in urllib. Except this one is compatible with Python 2. Kind of like how we do aio compat in aio_compat.py

---
In reply to: [399724837](https://github.com/Azure/azure-iot-sdk-python/pull/514#discussion_r399724837) [](ancestors = 399724837)